### PR TITLE
unit.d directory should be purged of unmanaged dropin files

### PR DESCRIPTION
**Pull request by treydock**

### CODE REVIEW

**Comment by ekohl** [comment]
Should this be configurable?

**Comment by raphink** [comment]
Making it configurable would be good; however I wonder where the parameter should reside.

**Comment by ekohl** [comment]
On the main systemd class and then add a parameter on this class that defaults to the value on the systemd class? That's the only thing I can think of because of the ```ensure_resource```.

**Comment by raphink** [comment]
I :+1: the parameter on the systemd class, but the current code is not a class, it's a define, so it should just use the value from the systemd class and not expose anything.

**Comment by ekohl** [comment]
@raphink not exposing a parameter might even be better since the ```include ::systemd``` is inside the class so you're only guaranteed to have the value after that.

**Comment by treydock** [comment]
@ekohl @raphink Parameter added and updated unit tests too.  The default is to purge, can change if needed.

File: manifests/dropin_file.pp [hunk 1]
```diff
@@ -50,9 +50,11 @@
 
   if $ensure != 'absent' {
     ensure_resource('file', "${path}/${unit}.d", {
-      ensure => directory,
-      owner  => 'root',
-      group  => 'root',
+      ensure  => directory,
+      owner   => 'root',
+      group   => 'root',
+      recurse => $::systemd::purge_dropin_dirs,
```
Comment (ekohl) [inline manifests/dropin_file.pp]: Is recursing needed? I thought purge would be enough.

Comment (treydock) [reply manifests/dropin_file.pp]: @ekohl I think recurse is needed, at least that's my read of docs for both purge and recurse.  I use that pattern in my profiles where I'm essentially doing what this define does, it works as expected though recurse does seem unnecessary at first.

https://docs.puppet.com/puppet/latest/type.html#file-attribute-purge

**Comment by treydock** [comment]
PR rebased

File: manifests/dropin_file.pp (hunk 1 above)
Comment (bastelfreak) [reply manifests/dropin_file.pp]: this is maybe a good candidate for an acceptance test?

File: manifests/init.pp [hunk 2]
```diff
@@ -42,6 +42,7 @@
   Enum['stopped','running']        $timesyncd_ensure = 'running',
   Optional[Variant[Array,String]] $ntp_server          = undef,
   Optional[Variant[Array,String]] $fallback_ntp_server = undef,
+  Boolean                         $purge_dropin_dirs = true,
```
Comment (bastelfreak) [inline manifests/init.pp]: can you check that every parameter is aligned at `=`?

**Comment by bastelfreak** [comment]
Hey @treydock, sorry for the slow response here :(
Can you rebase again please?

**Comment by treydock** [comment]
@bastelfreak rebased.  This module has no acceptance tests, should I add some in this pull request?

**Pull request merged**